141617737 edit foods aw

wlffann opened the pull request:
This PR begins the edit and save food stories. It currently will only work for the food name `<td>`, but it illustrates the desired effect. Updating it for the calories should just be a matter of DOM traversal, but I wanted to get some more eyes on this before I move forward. It seems like there must be a better way.

I think we might want to address the way we are saving objects in localStorage. Currently, we have a key of the food's name and the html for the food's `<tr>`. This means that the entire object needs to change if the food's name changes. Also, it makes it difficult to access the calorie value. It looks like we'll need to do some calculations later with these numbers, so exposing them now might be advantageous. We can create an object that looks more like `{"banana": {"name": "Banana", "calories": "35"}}`. This still doesn't change our naming problem, though...

[#141617737], [##141617739]
@case-eee

wlffann committed:
Add inline editing to foods table

This updates the table to feature inline editing of foods. Currently
not hooked up to local storage, but it is updating the DOM elements
[#141617737]

wlffann committed:
Update input styling

This updates the input styling to better match the styles of the table.
Makes the effect a little nicer
[#141617737]

wlffann committed:
Add localStorage save to update

This updates my previous inline form to update the object in
localStorage.
[#141617737]

case-eee commented:
@wlffann this is a great start, but you're right - there is a simpler way. let's just store what we need in localStorage aka the bare minimum. the html doesn't need to persist and live in localStorage, we can always construct that 👍 also - what about storing something like JSON in local storage? or a stringified version of JSON? that way, our data will be very organized and easy to access!

we'll be talking about this first thing tomorrow morning too 🎉

DavidKnott commented on lib/edit-foods.js:
```diff
@@ -0,0 +1,14 @@
+$('document').ready( function() {
+  $('body').on('click', '.food-row td', function() { 
+    var $el = $(this);
```
@wlffann Could you use a different variable name "el" is a little confusing to me." Never mind, you already changed it in another commit.

DavidKnott commented on lib/edit-foods.js:
```diff
@@ -1,14 +1,23 @@
+                <td class='food-name'>` + $input.val() + `</td>
+                <td class='food-calories'>` + $input.siblings().first().text()  + `</td>
+                <td class='food-delete'><button>-</button></td>
+              </tr>`); 
```
It'd be great if you could extract the food row into its own function so that you're passing a foodRow variable into "setItem".

wlffann replied:
I agree that would be a lot more clean. Soon, we won't be passing in this whole mess, thankfully. Once we have localStorage update, I'll move the new logic into a variable.

case-eee commented:
@wlffann let me know if after our class this morning, you'd like to talk through any of this 👍

wlffann commented:
@case-eee Thanks! Jason is working on our localStorage set up and we talked through it and feel much clearer after class today.

case-eee commented:
@wlffann that's great to hear 👍

wlffann committed:
Update edit food name to use localStorage

This changes the inline form for updating food names to update the
objects in local storage.
[#141617737]

wlffann committed:
Add edit ability to food calories

[#141617737]

wlffann committed:
Refactor edit function

This commit removes the duplicate event bindings to use just one. It
also pulls out some logic into stand alone functions.

[#141617737]

DavidKnott commented on lib/edit-foods.js:
```diff
@@ -0,0 +1,48 @@
+function pullOutEditedFood(currentFoods,food){
+  return $.grep(currentFoods, function(notFood) {
+    return notFood != food;
+  });
```
I'm so glad we figured out the double returns!

jdconrad89 replied:
plus one to what @DavidKnott said!

case-eee replied:
this is definitely tricky y'all. good work 👍

DavidKnott commented:
Looks ready to merge

jdconrad89 commented on lib/edit-foods.js:
```diff
@@ -0,0 +1,48 @@
+    food.calories = newData
+  }
+};
+
```
Love how we were able to get the update calories and food name into the single method!

jdconrad89 commented:
@wlffann everything looks great!
